db/migrate/20190920102446_clean_custom_values.rb

```diff
@@ -0,0 +1,22 @@
+class CleanCustomValues < ActiveRecord::Migration[5.2]
+  def up
+    invalid_cv = CustomValue
+      .joins(:custom_field)
+      .where("#{CustomField.table_name}.field_format = 'list'")
+      .where("value IS NOT NULL and value !~ '^[0-9]+$'")
```

Comment:
This will also return empty values .e.g

```
CustomValue.joins(:custom_field).where("#{CustomField.table_name}.field_format = 'list'").where("value IS NOT NULL and value !~ '^[0-9]+$'")
D, [2019-09-20T15:03:07.653475 #25414] DEBUG -- :   CustomValue Load (1.0ms)  SELECT "custom_values".* FROM "custom_values" INNER JOIN "custom_fields" ON "custom_fields"."id" = "custom_values"."custom_field_id" WHERE (custom_fields.field_format = 'list') AND (value IS NOT NULL and value !~ '^[0-9]+$')
=> [#<CustomValue:0x00007fca08c6b608 id: 140, customized_type: "WorkPackage", customized_id: 120, custom_field_id: 6, value: "">]
```

You could use:
```
CustomValue
  .joins(:custom_field)
  .where(custom_fields: { field_format: 'list'} )
  .where.not(value: '')
  .where("value !~ '^[0-9]+$'")
```

Reply:
Are empty values allowed? I see both NULL and empty values on my databases. Shouldn't we rather turn all empty values into nulls then ?

Reply:
Yes, both seem to be allowed (empty and NULL). Values for new custom values are assigned a `nil` by default and most front end forms will pass a `''` if no value is added. I have seen no trace of a cleanup taking place.